src/TextInput.js

```diff
@@ -1,167 +1,149 @@
-import React, { Component } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
+const TextInput = props => {
+  const [id, setId] = useState(props.id || idgen());
 
-    this.id = props.id || idgen();
+  const [idSetable, setIdSetable] = useState(true);
```

Comment:
not needed anymore?

Reply:
We are still using `props.id || idgen()` to set the id on line 8. Line 10 is used to ensure that the id can only be set once per mount.

Reply:
I'm not sure about this, it seems that neither `idSetable` nor `setIdSetable` are used anywhere right?

Reply:
You're absolutely right, it was removed in a previous edit. Also after thinking about it some more I understand why the associated logic was unnecessary.